src/main/java/org/reactome/orthoinference/EventsInferrer.java

```diff
@@ -334,4 +345,33 @@ private static void setInstanceEdits(int personId) throws Exception
 		EWASInferrer.setInstanceEdit(instanceEditInst);
 		PathwaysInferrer.setInstanceEdit(instanceEditInst);
 	}
+
+	/**
+	 * Download the Wormbase file that contains the WBGene IDs mapped to gene names, and then create a mapping of IDs to names.
+	 * @param wormbaseUrl -- URL to Wormbase file to be downloaded and processed.
+	 * @return -- Map<String, List,<String>> of WBGene IDs to gene names.
+	 * @throws IOException
+	 */
+	private static Map<String, List<String>> downloadAndProcessWormbaseFile(URL wormbaseUrl) throws IOException {
+		File wormbaseFile = Paths.get(wormbaseUrl.toString()).getFileName().toFile();
+		Map<String, List<String>> wormbaseMappings = new HashMap<>();
+		if (!wormbaseFile.exists()) {
+			FileUtils.copyURLToFile(wormbaseUrl, wormbaseFile);
+		}
```

Comment:
Does this block stop an already downloaded wormbase file from being redownloaded?  If so, are there times when re-downloading the file would be useful so it's guaranteed to be the most recent data?

Reply:
It does stop it from being redownloaded, but your right that it should redownload if the existing file is x number of days old

Reply:
Does the file take long to download?  Maybe it's just simpler downloading fresh each time if it's a small file.  If not, you could try implementing a check to see if the file has changed.  I did some quick research and you could check the HTTP header "if modified since" to determine if the file has been updated on Wormbase's server.

Reply:
File is now downloaded each time, and deleted once the mapping has been created